code/web/src/modules/crate/Item.js

```diff
@@ -25,52 +27,64 @@ class Item extends PureComponent {
     super(props)
 
     this.state = {
-      isLoading: false
+      isLoading: false,
+      surveyRedirect: false
     }
   }
 
   handleClick = (crateId) => {
     if (!this.props.user.style) {
-      this.props.history.push(styleSurvey.survey.path)
+      this.setState({ 
+        surveyRedirect: true
+      }) 
+      // this.props.history.push(styleSurvey.survey.path)
     } else {
       this.subscribeUserToCrate(crateId)
     }
   }
 
-  subscribeUserToCrate = (crateId) => {
-    this.setState({
+  static subscribeUserToCrate = (crateId, owner = this) => {
+    owner.setState({
```

Comment:
@GreysonElkins I'm curious about the use of `owner` instead of `this`. I haven't seen it before so I'd love to learn the reasoning behind it?

Reply:
@NakiNorton For sure! When importing `subscribeUserToCrate` into the Survey component for the submit button, `this` was still pointing to the `Item` component, which hadn't been instantiated. By adding `this` as a default value for a param, `Item` can still call the function as it always did (without declaring `this` as an argument), but this allows us to pass in a new `this` (which refers instead to the `Survey` component) as the object to affect state with.

Reply:
Ohh right! That's very clever. Thanks for the detailed explanation, much appreciated!!